middleware/fastcgi/fastcgi.go

```diff
@@ -166,6 +166,12 @@ func (h Handler) buildEnv(r *http.Request, rule Rule, fpath string) (map[string]
 		scriptFilename = absPath
 	}
 
+	// Strip PATH_INFO from SCRIPT_NAME
+	indexPathInfo := strings.LastIndex(scriptName, pathInfo)
+	if indexPathInfo != -1 {
+		scriptName = scriptName[:indexPathInfo]
+	}
+
```

Comment:
Maybe we want to use [strings.TrimPrefix](http://golang.org/pkg/strings/#TrimPrefix) here.